osh/hub/scan/mock.py

```diff
@@ -0,0 +1,133 @@
+        return
+
+    # get build tag name (build tag seems to always contain the release version)
+    return koji_proxy.getBuildTarget(target)['build_tag_name']
+
+
+def _get_build_arches(koji_proxy, task_id):
+    """
+    returns all arches available for given build
+    """
+    arches = []
+    for child_id in koji_proxy.getTaskDescendents(task_id, request=True):
+        child = koji_proxy.getTaskInfo(child_id, request=True)
+        if child['method'] == 'buildArch' and child['arch'] != 'noarch':
+            arches.append(child['arch'])
+
+    # noarch tasks can be built anywhere
+    return arches or koji_proxy.getAllArches()
+
+
+def _get_tag_specific_config(target):
+    """
+    returns build tag specific mock configuration which should be appended to
+    the generated mock config
+    """
+    # Fedora
+    match = re.search(r'f(\d+)', target)
+    if match is not None:
+        return f"config_opts['releasever'] = '{match[1]}'\n" \
+               f"config_opts['bootstrap_image'] = 'registry.fedoraproject.org/fedora:{match[1]}'\n"
+
+    # CentOS Stream
+    match = re.search(r'c(\d+)s', target)
+    if match is not None:
+        return f"config_opts['releasever'] = '{match[1]}'\n" \
+               f"config_opts['bootstrap_image'] = 'quay.io/centos/centos:stream{match[1]}'\n"
+
+    # RHEL
+    match = re.search(r'rhel-?(\d+)', target)
+    if match is not None:
+        return f"config_opts['releasever'] = '{match[1]}'\n" \
+               f"config_opts['bootstrap_image'] = 'registry.access.redhat.com/ubi{match[1]}/ubi'\n"
+
+    # fallback
+    return "config_opts['use_bootstrap_image'] = False\n"
+
+
+def generate_mock_configs(nvr, koji_profile, task_dir):
+    # retrieve the build
+    cfg = koji.read_config(koji_profile)
+    koji_proxy = koji.ClientSession(cfg['server'])
+    build = koji_proxy.getBuild(nvr)
+
+    # retrieve task parameters
+    task = koji_proxy.getTaskInfo(build['task_id'], request=True)
+    method = task['method']
+    params = koji.parse_task_params(method, task['request'])
+
+    # parse build tag name from task parameters
+    if method == 'build':
+        tag = _get_build_method_build_tag(koji_proxy, params['target'])
+    elif method == 'wrapperRPM':
+        tag = params['build_target']['build_tag_name']
+    else:
+        # FIXME: add actual error handling
+        logger.error(f'No build target for "{nvr}" available!')
+        return
+
+    # generate a config for every built arch
+    logger.debug(f'Generating mock configs for build tag "{tag}"')
+    for arch in _get_build_arches(koji_proxy, task['id']):
+        p = subprocess.run(['koji', '-p', koji_profile, 'mock-config',
+                            '--arch', arch, '--latest', '--tag', tag],
+                           check=True, stdout=subprocess.PIPE)
+        contents = p.stdout.decode()
+
+        # add extra repos
+        matched_repos = [repo for regex, repo
+                         in settings.MOCK_AUTO_EXTRA_REPOS.items()
```

Comment:
Could we please make it default to an empty dict so that it does not traceback in case the code is updated but local changes in `settings_local.py` are not yet merged?
```py
% osh-cli mock-build --config=auto --nvr curl-7.87.0-7.fc38
Traceback (most recent call last):
  File "/usr/bin/osh-cli", line 85, in <module>
    main()
  File "/usr/bin/osh-cli", line 78, in main
    parser.run()
  File "/usr/lib/python3.11/site-packages/kobo/cli.py", line 296, in run
    cmd.run(*cmd_args, **cmd_kwargs)
  File "/usr/lib/python3.11/site-packages/osh/client/commands/cmd_build.py", line 175, in run
    handle_perm_denied(e, self.parser)
  File "/usr/lib/python3.11/site-packages/osh/client/commands/cmd_build.py", line 173, in run
    task_id = self.submit_task(options)
              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/osh/client/commands/cmd_mock_build.py", line 20, in submit_task
    return self.hub.scan.mock_build(options['mock_config'],
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/xmlrpc/client.py", line 1122, in __call__
    return self.__send(self.__name, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/xmlrpc/client.py", line 1464, in __request
    response = self.__transport.request(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/kobo/xmlrpc.py", line 477, in request
    result = transport_class.request(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/xmlrpc/client.py", line 1166, in request
    return self.single_request(host, handler, request_body, verbose)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/kobo/xmlrpc.py", line 391, in _single_request3
    return self.parse_response(response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/xmlrpc/client.py", line 1354, in parse_response
    return u.close()
           ^^^^^^^^^
  File "/usr/lib64/python3.11/xmlrpc/client.py", line 668, in close
    raise Fault(**self._stack[0])
xmlrpc.client.Fault: <Fault 1: AttributeError: 'Settings' object has no attribute 'MOCK_AUTO_EXTRA_REPOS'>
```

Reply:
And what about doing it the other way around?  First contribute you downstream settings to `osh-rh-conf` and then do the update.

Reply:
Technically, it can be done.  My impression is that handling of configuration files in other projects that I use does not require each supported option to be present in each configuration file.  When an option has a reasonable default value, the application should simply use the default value when the option is missing in a config file.

Reply:
Ok, I'll move the default to `settings.py`.

Reply:
@lzaoral Please put there at least the fallback to an empty dict.  Even if we update `osh-rh-conf`, the changes need to be manually merged because of [OSH-387](https://issues.redhat.com/browse/OSH-387).

Reply:
Addressed in 75eb823.